samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -260,6 +263,12 @@ protected void setupSystem() {
 
     setSerde(log4jSystemConfig, systemName, streamName);
 
+    // Explicitly create stream appender stream with the partition count the same as the number of containers.
+    JobConfig jobConfig = new JobConfig(config);
+    StreamSpec streamSpec = StreamSpec.createStreamAppenderSpec(streamName, systemName, jobConfig.getContainerCount());
```

Comment:
What's the behavior for standalone mode, which doesn't use a predefined container count?
Perhaps use the number of tasks in that case? Or require a config. Or use some reasonable default.

Reply:
Good point. I looked into using the number of tasks, but it involves fetching metadata for each input stream's system admin to figure out the max task count. Is there an easier way to get task count? If not, I think I'll opt to require a config with a default of 8. That sound good?

Reply:
In digging further, the stream appender log stream is keyed by container name, so it looks like the number of partitions should be determined by container count. For standalone, since the number of containers cannot be determined during start up, requiring a config seems to be the only option.

On another note, I'm not sure how StreamAppender even works in standalone because the container names are only set by the `run-container.sh` and `run-jc.sh` scripts: `export JAVA_OPTS="$JAVA_OPTS -Dsamza.container.name=samza-container-$SAMZA_CONTAINER_ID"`. StreamAppender throws an exception if there is no container name set. Standalone mode probably needs more investigation.

Reply:
Thanks for looking into it. Perhaps rather than adding a new config for the container count, it would be more appropriate to add another log4j parameter, analogous to `StreamName` and make both mandatory for standalone.

Reply:
The `KafkaSystemAdmin.createStream` seems to hang on the ZK connection when called from StreamAppender when running it in `hello-samza`. Need to look into why.

Reply:
Issue was that the `master` branch of `samza` matches the `latest` branch of `hello-samza`. I was using the `master` branch of `hello-samza`. Seems to work now. Thanks @bharathkk for helping look into this.